### PR TITLE
Deduplicate frequent translation strings

### PR DESCRIPTION
**Pull request by denyskon**
As title. Cases where we get fancy with constructing the string are excluded for now, as are most strings which contain more detail, e. g. "Edit" is included but "Edit File" is not.

Per commit, only one string or few less frequent strings are changed to simplify review.

Part of #27700 , #24488

### CODE REVIEW

**Comment by denyskon** [comment]
I'm getting tired, this PR is getting too big, and I also think it can be done in multiple rounds, so let's start with this :)

**Comment by lafriks** [comment]
Did you check if deduplicated values are also the same in other languages? I have not checked yet but at least for Latvian language `Public` and `Private` translates differently depending on context (organization/user/repository) - for user and repository it will be `Publisks` but for organization - `Publiska`

I did not check all but there could be a similar problem with read/write and maybe others

**Comment by denyskon** [comment]
Honestly I don't think we should have to take care of every possible translation context of the words, it's impossible to track....

File: templates/explore/navbar.tmpl [hunk 1]
```diff
@@ -9,7 +9,7 @@
 			</a>
 		{{end}}
 		<a class="{{if .PageIsExploreOrganizations}}active {{end}}item" href="{{AppSubUrl}}/explore/organizations">
-			{{svg "octicon-organization"}} {{ctx.Locale.Tr "explore.organizations"}}
```
Comment (delvh) [inline templates/explore/navbar.tmpl:12]: From a personal viewpoint (#29816): Oh no, a merge conflict :smiling_face_with_tear:

**Comment by denyskon** [comment]
@silverwind What do you think?

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -1281,7 +1246,6 @@ editor.file_delete_success = File "%s" has been deleted.
 editor.name_your_file = Name your file…
 editor.filename_help = Add a directory by typing its name followed by a slash ('/'). Remove a directory by typing backspace at the beginning of the input field.
 editor.or = or
-editor.cancel_lower = Cancel
```
Comment (delvh) [inline options/locale/locale_en-US.ini:1284]: How did a `lower` ever end up in the translations?
And why isn't it even lowercased?

**Review by delvh — APPROVED** [review]

**Comment by lafriks** [comment]
But that will make it untranslatable correctly and that defeats the whole point of translating. It would be the same as proposing to replace all occurrences in English language for `repositories` to just single `repository` it would just look plainly wrong

**Review by lafriks — CHANGES_REQUESTED** [review]
Blocking for now as this really needs to be reviewed more from the point of other languages

**Comment by denyskon** [comment]
But if we really want to make it correct from every grammatical point, we would need a new translation key for every scope where it is used. We have private in the context of users, orgs, repos, projects, email addresses, members.... we simply cannot maintain an own translation for every single one of these things. What would be the alternative? We cannot give a word a gender (that's the issue if I understand correctly) because it changes depending on the language.

Does Latvian for example really not contain a way to express this state neutrally? In Ukrainian or Russian there's the same problem, but you can avoid it by using an adverb instead of the adjective. Yes it sounds weird, but you can get used to it.

Alternatively you can translate with braces like `Publisk(a)` in your case.

**Comment by silverwind** [comment]
I will add some basic "linting" capabilities to https://github.com/go-gitea/gitea/pull/29825 which will be a good validation to run this PR against to detect unused or missing translations.

**Comment by lafriks** [comment]
> But if we really want to make it correct from every grammatical point, we would need a new translation key for every scope where it is used. We have private in the context of users, orgs, repos, projects, email addresses, members.... we simply cannot maintain an own translation for every single one of these things. What would be the alternative? We cannot give a word a gender (that's the issue if I understand correctly) because it changes depending on the language.
> 
> Does Latvian for example really not contain a way to express this state neutrally? In Ukrainian or Russian there's the same problem, but you can avoid it by using an adverb instead of the adjective. Yes it sounds weird, but you can get used to it.
> 
> Alternatively you can translate with braces like `Publisk(a)` in your case.

Latvian language does not have neutral way of expressing such cases and while it would be possible to write `Publisks(-a)` such form are pretty much never used because for some words it's not possible to do as not only last letter changes.

**Comment by silverwind** [comment]
> Latvian language does not have neutral way of expressing such cases and while it would be possible to write `Publisks(-a)` such form are pretty much never used because for some words it's not possible to do as not only last letter changes.

On a scale of 1 (nit) to 10 (meaning gets lost), how bad is the error? Certain minor issues have to be accepted imho.

**Review by silverwind — APPROVED** [review]

**Review by wxiaoguang — CHANGES_REQUESTED** [review]
Different languages have different words, some words can't be simply "deduplicated"

The last time it was wrong for "color":    Translation wrong for word color in PR #16647 #17384 & Re-separate the color translation strings #17390

**Comment by denyskon** [comment]
Then please propose a solution

**Comment by lafriks** [comment]
Imho adjectives most probably cannot be deduplicated to keep context, also cannot deduplicate where one is verb (action), other noun (label) even if both in English are written the same

**Comment by denyskon** [comment]
I agree with the second one, but we cannot create a new adjective translation for every case where it is used. If they exist currently, it's mostly just by accident

**Comment by denyskon** [comment]
Compromise for now - I revert the commits for everything that is used as an adjective, we can at least merge the non-confusing nouns, and then we think of a solution for the other ones

File: models/shared/types/ownertype.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ func (o OwnerType) LocaleString(locale translation.Locale) string {
 	case OwnerTypeRepository:
 		return locale.TrString("concept_code_repository")
 	case OwnerTypeOrganization:
-		return locale.TrString("concept_user_organization")
+		return locale.TrString("organization")
```
Comment (wxiaoguang) [inline models/shared/types/ownertype.go:26]: Why I used `concept_user_organization` here, the reason is the same as the previous discussion: I believe we should clearly define the concepts with context. I think "organization" could have different means in different contexts in different languages.

* One context, one concept, one translation.
* Different context, different concept, different translation.

Comment (denyskon) [reply models/shared/types/ownertype.go:26]: I increasingly think we need to restructure translations completely. Not group them by usage (new issue page, admin user table, installation page) but by context or meaning (`[permissions]`, `[operations]`, `[repo_units]` and so on)

**Comment by wxiaoguang** [comment]
I still think there are unclear concepts.

For example: "dashboard". I can see in Chinese translation, it was translated to "Home page". It is right in Chinese when the "dashboard" is used in the home page context. But it is wrong when it is used in the admin pages.

To answer the question "Then please propose a solution":

Only take actions when it matches at least one of these conditions:

1. If there is a real problem and it really affects end users (say, a bug).
1. There are some native speakers and language experts clearly know how to do it.
1. We clearly know all the details and know how to do it.

----

For this PR's change, I think it doesn't aim to resolve bugs, and we haven't got a full picture of all language translations.

**Comment by denyskon** [comment]
@go-gitea/technical-oversight-committee 

I think we need an opinion from the TOC here

**Comment by silverwind** [comment]
Imho some people just need to go through the modifications in the ini and highlight the cases where deduplication is not possible and then we split those up again. Just a normal review process imho. Just saying "no" to the entire PR is not constructive feedback.

**Comment by wxiaoguang** [comment]
> Imho some people just need to go through the modifications in the ini and highlight the cases where deduplication is not possible and then we split those up again. Just a normal review process imho. Just saying "no" to the entire PR is not constructive feedback.

I'd like to, but I only understand English and Chinese,  and I have shown the real problem (eg: "dashboard"). So I can't comment for others. Actually I would prefer to insist my suggestion in https://github.com/go-gitea/gitea/pull/29823#issuecomment-2001953326 .

**Comment by jolheiser** [comment]
We may need wider feedback on this, or to take inspiration from other projects. 
I agree with the notion of reducing translation complexity, but also know that it's never as simple as deduping. 

Do any of our translators have a suggestion on the matter, how it could be made easier? 
If it's to reorganize them by context, like mentioned above?

Would weblate make this easier or harder?

Glad this PR sparked the conversation, but we'll want to make sure we have an agreed solution before we merge.

**Pull request closed without merging**